test/metabase/util/log_test.clj

```diff
@@ -48,3 +48,28 @@
 
     (is (empty? (get-context))
         "Context should be reset after exception")))
+
+(deftest ^:parallel with-error-context-works
+  (testing "has correct data for a basic example"
+    (is (= {:foo "bar"}
+           (log/get-exception-data
+            (try (log/with-context {:foo "bar"}
+                   (/ 1 0))
+                 (catch Exception e
+                   e))))))
+  (testing "nested: inner overrides outer"
+    (is (= {:data "inner"}
+           (log/get-exception-data
+            (try (log/with-context {:data "outer"}
+                   (log/with-context {:data "inner"}
```

Comment:
Silently overwriting this feels kind of error prone.

We don't need to fix it here, I'm not sure what's best but what do you think of either

1. combining context key collisions instead of overwriting
2. warning when there's a collision